Add SolidWorks navigation style

theepicviolin opened the pull request:
Addresses https://github.com/FreeCAD/FreeCAD/issues/12021

Doesn't include the SolidWorks Alt+MMB ["Roll View"](https://help.solidworks.com/2024/English/SolidWorks/sldworks/t_roll_view.htm?id=a172b767fd314359b2ed52b978d273f3#Pg0), but includes [Rotate](https://help.solidworks.com/2024/English/SolidWorks/sldworks/t_rotate_view.htm?id=2354dc29a7df4dfbaf4eaab8062be627#Pg0) as MMB, [Pan](https://help.solidworks.com/2024/English/SolidWorks/sldworks/t_pan_fundamentals.htm?id=a8f401062c1a47c8a3a0ab408fc6f4b1#Pg0) as Ctrl+MMB, and [Zoom](https://help.solidworks.com/2024/English/SolidWorks/sldworks/t_zoom_in_out.htm?id=c78a4f4a4c9348718f8acdc920038f05#Pg0) as Shift+MMB. 
(it's just a copy of the Blender nav style but with Zoom and Pan swapped) 

Named as SW to avoid having a trade name, as noted in the thread.

maxwxyz commented:
@Rexbas could you test this?

Rexbas commented:
I will test this.

Rexbas reviewed (changes requested):
I believe the navigation itself works. However, the option to select the style should also be added in the bottom right corner of the screen because it is not there right now. 
![image](https://github.com/user-attachments/assets/0c137f78-d8f5-42e2-804e-0142e7d43dff)

If I am not mistaken, you can add it somewhere in this file: https://github.com/FreeCAD/FreeCAD/blob/main/src/Mod/Tux/NavigationIndicatorGui.py

MisterMakerNL commented:
Or just replace CAD.

theepicviolin commented:
@Rexbas I just pushed a commit to add it to the list, and added new icons for it too, to match with the other styles. 

@tritao makes a good point that there's already Maya and Revit as commercial product names, so I renamed it to SolidWorks, to make it extra clear to users.

maxwxyz commented:
@FreeCAD/design-working-group any feedback for the provided SW icon or any other proposals?
Not sure if the included text renders well at small size
![grafik](https://github.com/user-attachments/assets/0f501527-22b6-4772-93f8-90b0a98709f3)

@chennes is it ok to have other commercial CAD names as reference in FreeCAD for navigation or is this copyrighted (SolidWorks, Inventor, Maya,...)

Rexbas commented:
> Not sure if the included text renders well at small size

The text in the icon is definitely not readable, just like the TinkerCAD and OpenSCAD icons. And the minor differences between all other mouse icons are not really visible either.

![image](https://github.com/user-attachments/assets/5d67cf14-47f4-45b4-a841-6e23bbafa911)

Maybe we should use the icon to indicate what kind of navigation style it is. So the gesture icon for Gesture and MayaGesture navigation style to indicate that it is a gesture navigation style. A touchpad icon for the Touchpad navigation style to indicate it is for touchpad use and all other styles should then have the mouse icon from the CAD style to indicate that it is designed to be used with a mouse. This is however probably something for a separate PR.

From a technical perspective I can approve this PR. As for the name copyright issue, I leave this to others to decide.

pierreporte commented:
Agree with @Rexbas. The icons aren’t very useful but they could simply indicate he type of navigation style: mouse, gestures, touchpad, etc.

MisterMakerNL commented:
![solidworks-logo](https://github.com/user-attachments/assets/89b98254-c81d-41ad-89db-72b08059ab3d)
Here a sw logo in svg!

maxwxyz commented:
I'm really hesitant using these things as "CAD competitor" which are probably trademarked

Syres916 commented:
@WandererFan do new nav styles not have to be added to `TechDraw` as well now like TouchPad has `QGVNavStyleTouchpad.cpp` and `QGVNavStyleTouchpad.h` or is this purely optional?

MisterMakerNL commented:
> I'm really hesitant using these things as "CAD competitor" which are probably trademarked

It's just a tiny logo in the corner, and it doesn't use the same colors.

kadet1090 commented:
That is stepping on thin ice, I don't like this idea also. The logo is clearly theirs, they can even have logo usage guidelines forbidding making the logo grayscale in that way.

MisterMakerNL commented:
So I looked at it again, and there is no way that you only encounter the logo without the text, so there isn't even any point for the logo. Half of the logo's are pointless and not even recognizable.
So tldr original logo is fine.

WandererFan commented:
@Syres - yes, a QGVNavStyleSolidWorks.cpp will have to be created to translate the appropriate keystrokes/mouse actions into QGraphicsScene actions.

theepicviolin commented:
@Syres916 @WandererFan Thanks for letting me know about `TechDraw`, I just added another commit for that. 

I'm generally in agreement that the icons for the nav styles aren't super helpful as is, and I agree too that it's fine to leave it for now.

hyarion commented:
I'm not a lawyer but referring to trademarks to state compatibility or likeness is usually ok - as long as little as possible of the trade mark is used.

In us this is called "nominative fair use" and similar exists elsewhere too (permissible descriptive use).

Logo: probably not ok
Word mark: probably ok in this context

If we want to be on the safe side, we could append a "-like".

hkienle commented:
> I'm not a lawyer but referring to trademarks to state compatibility or likeness is usually ok - as long as little as possible of the trade mark is used.
> 
> In us this is called "nominative fair use"

I am also not a lawyer, but I also believe that trademark law allows such uses. Another related concept is "descriptive use" of a trademark. (If you have a car repair shop, it is ok to have a sign "Fixing Fords", say.)
